packages/jest-runtime/src/index.ts

```diff
@@ -758,6 +757,20 @@ class Runtime {
           `You have requested '${globalVariable}' as a global variable, but it was not present. Please check your config or your global environment.`,
         );
       }),
+    ];
+
+    const injectedModuleArguments = this.constructInjectedModuleArguments();
+
+    if (injectedModuleArguments.length !== args.length) {
+      throw new ReferenceError(
+        `We are injecting ${args.length} arguments into module wrapper, expected ${injectedModuleArguments.length}.\n\nThis is a bug in Jest, please report it`,
```

Comment:
Ideally this should be a test, so we could avoid a runtime check. But maybe it's worth to have a small safety net? The overhead seems negligible

Reply:
I'm unable to detect what the string is. My only idea is using `arguments.callee()`, but that won't work in strict mode. I guess I could write some test that mocks out the `transform` call and inspect the args?

Reply:
I can also just cheat and call the private function 😀

Reply:
yeah, did that 👍

Reply:
Running a script is a side effect already, so I guess we don't have too much of a choice without extracting args creation to a separate function?

Reply:
yeah, think so. want me to remove the length check?

Reply:
Let's do it